WeatherStationClient.py

```diff
@@ -28,8 +76,984 @@ def __init__(self, code, response):
         self.code = code
         self.response = response
 
+    @classmethod
+    def _get_specific(cls, code, response):
+        specific_type = ResponseOK if code == 200\
+                   else ResponseNoContent if code == 204\
+                   else cls
+        return specific_type(code, response)
+
+
+class ApiBoolResponse(ApiResponse):
+    @classmethod
+    def _from_response(cls, response):
+        class _BoolResponse(cls, type(response)):
+            def __init__(self, value):
+                self._value = value
+                type(response).__init__(self, response.code, response.response)
+            def __bool__(self):
+                return self._value
+        return _BoolResponse
+
+
+class ResponseOK(ApiResponse):
+    pass
+
+
+class ResponseNoContent(ApiResponse):
+    pass
+
+
+class ListResponse(list):
```

Comment:
Is this meta-meta-meta programming really need? Aren't params that's used as input to this machinery as long or in common approximation long as setting model attributes explicitly?

Reply:
Wiem. Też mi się to nie podoba.

Na swoją obronę mam to, że za każdym raziem, kiedy produkowałem tego rodzaju klasę czy metodę, to miałem konkretne kawałki powtarzalnego kodu, który napisałem, a którą to powtarzalność chciałem usunąć. Dlatego wydaje mi się, że mogę wytłumaczyć się z każdej podobnej meta-meta-klasy/metody.

Jeśli masz pomysł, jak obniżyć stopień skomplikowania mojego kodu bez wprowadzania powtarzalności: chętnie wysłucham.

W tym konkretnym wypadku celem istnienia `ListResponse` jest zapewnienie, by user - w odpowiedzi na zapytanie w rodzaju `GET /user/stations` czyli `user.get_stations` otrzymał obiekt, który może być przezeń równie dobrze traktowany jako `ApiResponse` jak i jako lista modeli. Właśnie po to, by umożliwić to, czego (słusznie) życzyłeś sobie powyżej. W związku z tym obiekt zwrócony przez `user.get_stations` winien dziedziczyć zarówno po `list`, jak i po `ApiResponse` - i to najchętniej nie po `ApiResponse`, ile po `ResponseOK` albo `ResponseNoContent`, w zależności od tego, co zostało zwrócone. 

Chociaż tutaj akurat może rzeczywiście przesadziłem z rozróżnieniem między `ResponseOK` a `ResponseNoContent`. Może wystarczy po prostu `ApiResponse`, bez tych podklas. Jednak zdało mi się, że akurat wprowadzenie tych podklas wprowadza (relatywnie) na tyle niewielki stopień skomplikowania, że w sumie czemu nie. `Model._bind` na przykład jest już gorszy, ale także (IMO) potrzebniejszy.